azblob/sas_service.go

```diff
@@ -20,11 +20,20 @@ type BlobSASSignatureValues struct {
 	Identifier         string  `param:"si"`
 	ContainerName      string
 	BlobName           string // Use "" to create a Container SAS
+	Directory          string // Not nil for a directory SAS (ie sr=d)
```

Comment:
Can we do struct embedding here to create a separate struct for directory sas signature?
```
type DirectorySASSignatureValues struct {
    BlobSASSignatureValues
    Directory
}
```

I'm a little conflicted here because  `list` is not a valid permission for BlobSAS and having a different struct for directory SAS makes the use-case clearer.

Reply:
Directories are represented as blobs over the blob endpoint so I'm a bit conflicted here as well.

Please check what the other SDKs are doing and we can mimic their structure. There has been improvements on SAS generation in the other SDKs so things may look very different.

Reply:
Here are couple of links which might be relevant - 

[SAS Generation in Blob Python SDK](https://github.com/Azure/azure-sdk-for-python/blob/e4e6c58d625667f24a57ceacbf49b77585717bb2/sdk/storage/azure-storage-blob/azure/storage/blob/_shared_access_signature.py#L129)

[SAS Generation in Data-lake Python SDK](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_shared_access_signature.py)